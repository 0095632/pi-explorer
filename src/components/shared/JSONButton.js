import React from 'react'
import PropTypes from 'prop-types'

const JSONButton = props =>
  <a href={props.urlFn(props.id)} type="application/hal+json">
    <span
      style={{
        fontSize: '0.8em',
        backgroundColor: 'white',
        borderRadius: '3px',
        color: 'black',
        padding: 3,
        marginLeft: 10,
      }}
    >
      JSON
    </span>
  </a>

JSONButton.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  urlFn: PropTypes.func.isRequired,
}

export default JSONButton
