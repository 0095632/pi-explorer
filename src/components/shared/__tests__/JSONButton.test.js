import React from 'react'
import {shallow} from 'enzyme'

import JSONButton from '../JSONButton'

it('renders button with given url', () => {
  const url = `https://somebackend.xyz/resource/12345`
  const btn = shallow(<JSONButton url={url} />)
  expect(btn.find('BadgeButton').props().url).toEqual(url)
  expect(btn.getNodes()).toMatchSnapshot()
})
