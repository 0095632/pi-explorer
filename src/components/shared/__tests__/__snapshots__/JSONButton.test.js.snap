// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with given url 1`] = `
Array [
  <a
    href="https://somebackend.xyz/resource/12345"
    type="application/hal+json"
>
    <span
        style={
            Object {
                "backgroundColor": "white",
                "borderRadius": "3px",
                "color": "black",
                "fontSize": "0.8em",
                "marginLeft": 10,
                "padding": 3,
              }
        }
    >
        JSON
    </span>
</a>,
]
`;
